### ISSUE TITLE
[BUG] structs.scad does not parse

### ISSUE THREAD

**Issue by tcurdt** [revarbat/BOSL2#167]
**Describe the bug**
structs.scad does not parse (for me)

**Code To Reproduce Bug**

Just opening the file, this gives me a syntax error:
https://github.com/revarbat/BOSL2/blob/master/structs.scad#L108

**Additional context**

- OpenSCAD Version:  2019.12.20 (git 2a943a57)

**Comment by tcurdt** [comment]
Also a problem with version 2020.05.17 (git b0dabf69)

**Comment by tcurdt** [comment]
This seems to work OK:
```
function is_struct(struct, ind=0) = let(dim = array_dim(struct))
    is_list(struct) && (
	struct == [] ||	
	len(dim)==2 && dim[1]==2
    );
```

**Comment by revarbat** [comment]
Argh.  Yes, OpenSCAD introduced a parsing bug recently that made having a `let()` immediately follow a logical operator like `||` or `&&` a syntax error, despite it being a valid expression prefix.  Putting the `let()` and its following expressions all in a paren `()` block should fix it.

**Comment by revarbat** [comment]
Fixed in d20caf7d59ff419b1cd59749b5447b1c335e9983
